### PR TITLE
[hardknott] recipes-core audit

### PR DESCRIPTION
**Pull request by SparkingSpork**
This PR includes audit changes to all `recipes-core` recipes **except** images and packagegroups which will be in follow-up PRs. I checked that all of these were still needed and filed work items for updates. 

AzDO Work Item: https://dev.azure.com/ni/DevCentral/_workitems/edit/2106907/

## Testing:
Built all recipes after changes.

### CODE REVIEW

File: recipes-core/busybox/busybox_1.33.%.bbappend [hunk 1]
```diff
@@ -1,50 +1,51 @@
...
 
-INITSCRIPT_NAME_${PN}-acpid = "busybox-acpid"
-INITSCRIPT_PARAMS_${PN}-acpid = "start 20 2 3 4 5 . stop 20 0 1 6 ."
+INITSCRIPT_NAME:${PN}-acpid = "busybox-acpid"
+INITSCRIPT_PARAMS:${PN}-acpid = "start 20 2 3 4 5 . stop 20 0 1 6 ."
 
 # Remove default busybox udhcpd init script; on NILRT images
 # udhcpd is invoked directly from ifplugd action scripts
+# unset line must be left as old syntax for now or it's unparsed.
 unset INITSCRIPT_NAME_${PN}-udhcpd
```
Comment (SparkingSpork) [inline recipes-core/busybox/busybox_1.33.%.bbappend]: Calling this out for specific review. I'm not sure why this didn't work with the new syntax.

Comment (amstewart) [reply recipes-core/busybox/busybox_1.33.%.bbappend]: That's interesting. I wonder if upstream forgot to accommodate the new syntax in the `unset` operation. :thinking:

File: recipes-core/distro-feed-configs/distro-feed-configs.bbappend [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
-RDEPENDS_${PN}-ptest += " bash "
+RDEPENDS:${PN}-ptest += " bash "
 
-do_install_ptest() {
-    install -d ${D}${PTEST_PATH}
-    install -m 755 ${WORKDIR}/test_feed_listings.sh ${D}${PTEST_PATH}
-    install -m 644 ${WORKDIR}/ptest-format.sh ${D}${PTEST_PATH}
-    install -m 755 ${WORKDIR}/run-ptest ${D}${PTEST_PATH}
+do_install:ptest() {
```
Comment (amstewart) [inline recipes-core/distro-feed-configs/distro-feed-configs.bbappend]: `do_install_ptest` isn't an override. [It's a task](https://github.com/ni/openembedded-core/blob/nilrt/master/hardknott/meta/classes/ptest.bbclass#L39) of the `ptest.bbclass`, and should use underscores.

File: recipes-core/initrdscripts/init-restore-mode_1.0.bb [hunk 3]
```diff
@@ -12,17 +12,17 @@ SRC_URI = "\
 	file://ni_provisioning.answers.default \
 "
 
-SRC_URI_append_xilinx-zynqhf = "\
+SRC_URI:append:xilinx-zynqhf = "\
 	file://disk_config_xilinx-zynqhf \
 "
```
Comment (amstewart) [inline recipes-core/initrdscripts/init-restore-mode_1.0.bb:17]: Do we have a tracking item to remove this?

File: recipes-core/util-linux/util-linux-nilrt.bb [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
+	file://test_hwclock.sh \
 "
 
 inherit ptest
 
-do_install_ptest() {
-    install -m 0644 ${WORKDIR}/ptest-format.sh ${D}${PTEST_PATH}
-    install -m 0755 ${WORKDIR}/run-ptest       ${D}${PTEST_PATH}
-    install -m 0755 ${WORKDIR}/test_hwclock.sh ${D}${PTEST_PATH}
+do_install:ptest() {
```
Comment (amstewart) [inline recipes-core/util-linux/util-linux-nilrt.bb]: do_install_ptest

**Review by amstewart — CHANGES_REQUESTED** [review]

File: recipes-core/busybox/busybox_1.33.%.bbappend (hunk 1 above)
Comment (amstewart) [reply recipes-core/busybox/busybox_1.33.%.bbappend]: Either way, I'm fine with leaving it in the old syntax until we get to the kirkstone rebase.

File: recipes-core/initrdscripts/init-restore-mode_1.0.bb (hunk 3 above)
Comment (SparkingSpork) [reply recipes-core/initrdscripts/init-restore-mode_1.0.bb:17]: Yep! https://dev.azure.com/ni/DevCentral/_workitems/edit/2112096

File: recipes-core/busybox/busybox_1.33.%.bbappend (hunk 1 above)
Comment (SparkingSpork) [reply recipes-core/busybox/busybox_1.33.%.bbappend]: Created https://dev.azure.com/ni/DevCentral/_workitems/edit/2114544, will link the work item number here

**Review by amstewart — APPROVED** [review]

**Pull request merged**